packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -650,16 +651,22 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
             }
 
             if (!ssr) {
-              const url = rawUrl.replace(cleanUpRawUrlRE, '').trim()
+              // Remove comments and trim whitespace
+              const cleanedUrl = stripLiteral(rawUrl)
+              const match = cleanedUpRawUrlRE.exec(cleanedUrl)
+              const url = match
+                ? rawUrl.slice(match[1].length, match[2].length)
+                : rawUrl
               if (
                 !urlIsStringRE.test(url) ||
                 isExplicitImportRequired(url.slice(1, -1))
               ) {
                 needQueryInjectHelper = true
+                // Use rawUrl to avoid removing comments like @vite-ignore
                 str().overwrite(
                   start,
                   end,
-                  `__vite__injectQuery(${url}, 'import')`,
+                  `__vite__injectQuery(${rawUrl}, 'import')`,
```

Comment:
I think this should still work (as `//` comments should end up with a new line because they are wrapped in the `import(...)` call), and it is better to avoid removing the comments, for example in case other plugins are checking for `@vite-ignore`.

It is also interesting that `@vite-ignore` here can't be completely ignored, the import query needs to be injected dynamically even if it is used.